android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMKeyboardDownloaderActivity.java

```diff
@@ -384,6 +393,67 @@ public static boolean isCustom(String u) {
     return ret;
   }
 
+  // If a font JSONObject contains multiple font font files, only keep the .ttf source
+  private static void findTTF(JSONObject jsonFont) {
+    boolean updateJsonFont = false;
+    try {
+      JSONArray fontSource = jsonFont.optJSONArray(KMManager.KMKey_FontSource);
+      if (fontSource != null) {
+        boolean containsTTF = hasTTFFont(fontSource);
+        if (containsTTF) {
+          for (int i = fontSource.length() - 1; i >= 0; i--) {
+            String s = fontSource.getString(i);
+            if (!FileUtils.isTTFFont(s)) {
+              updateJsonFont = true;
+              if (Build.VERSION.SDK_INT > 19) {
+                fontSource.remove(i);
+              } else {
+                fontSource = removeJsonObjectAtIndex(fontSource, i);
+              }
+            }
+          }
+
+          if (updateJsonFont) {
+            JSONArray copy = fontSource;
+            jsonFont.remove(KMManager.KMKey_FontSource);
+            jsonFont.put(KMManager.KMKey_FontSource, copy);
+          }
+        }
+      }
+    } catch (JSONException e) {
```

Comment:
In what scenarios can you see an exception? Shouldn't we at least log it?

Reply:
The IDE forces me to enclose the JSON calls in try/catch. I can add logging